backend/FwLite/LcmCrdt.Tests/Changes/ChangeSerializationTests.cs

```diff
@@ -0,0 +1,108 @@
+        yield return [new RemoveComplexFormTypeChange(Guid.NewGuid(), Guid.NewGuid())];
+        yield return [new CreateComplexFormType(Guid.NewGuid(), new MultiString())];
+        yield return [new AddComplexFormTypeChange(Guid.NewGuid(), new() { Id = Guid.NewGuid(), Name = new()})];
+        yield return [SetComplexFormComponentChange.NewComplexForm(Guid.NewGuid(), Guid.NewGuid())];
+        yield return [SetComplexFormComponentChange.NewComponent(Guid.NewGuid(), Guid.NewGuid())];
+        yield return [SetComplexFormComponentChange.NewComponentSense(Guid.NewGuid(), Guid.NewGuid(), Guid.NewGuid())];
+
+        yield return [new JsonPatchChange<Entry>(Guid.NewGuid(), new JsonPatchDocument<Entry>())];
+        yield return [new JsonPatchChange<Sense>(Guid.NewGuid(), new JsonPatchDocument<Sense>())];
+        yield return [new JsonPatchChange<ExampleSentence>(Guid.NewGuid(), new JsonPatchDocument<ExampleSentence>())];
+        yield return [new JsonPatchChange<WritingSystem>(Guid.NewGuid(), new JsonPatchDocument<WritingSystem>())];
+        yield return [new JsonPatchChange<PartOfSpeech>(Guid.NewGuid(), new JsonPatchDocument<PartOfSpeech>())];
+        yield return [new JsonPatchChange<SemanticDomain>(Guid.NewGuid(), new JsonPatchDocument<SemanticDomain>())];
+        yield return [new JsonPatchChange<ComplexFormType>(Guid.NewGuid(), new JsonPatchDocument<ComplexFormType>())];
+        yield return [new DeleteChange<Entry>(Guid.NewGuid())];
+        yield return [new DeleteChange<Sense>(Guid.NewGuid())];
+        yield return [new DeleteChange<ExampleSentence>(Guid.NewGuid())];
+        yield return [new DeleteChange<WritingSystem>(Guid.NewGuid())];
+        yield return [new DeleteChange<PartOfSpeech>(Guid.NewGuid())];
+        yield return [new DeleteChange<SemanticDomain>(Guid.NewGuid())];
+        yield return [new DeleteChange<ComplexFormType>(Guid.NewGuid())];
+        yield return [new DeleteChange<ComplexFormComponent>(Guid.NewGuid())];
+    }
+
+    [Theory]
+    [MemberData(nameof(Changes))]
+    public void CanRoundTripChanges(IChange change)
+    {
+        var config = new CrdtConfig();
+        LcmCrdtKernel.ConfigureCrdt(config);
+        var type = change.GetType();
+        var json = JsonSerializer.Serialize(change, config.JsonSerializerOptions);
+        var newChange = JsonSerializer.Deserialize(json, type, config.JsonSerializerOptions);
+        newChange.Should().BeEquivalentTo(change);
+    }
+
+    [Fact]
+    public void ChangesIncludesAllValidChangeTypes()
+    {
+        var allChangeTypes = LcmCrdtKernel.AllChangeTypes();
+        allChangeTypes.Should().NotBeEmpty();
+        var testedTypes = Changes().Select(c => c[0].GetType()).ToArray();
+        using (new AssertionScope())
+        {
+            foreach (var allChangeType in allChangeTypes)
+            {
+                testedTypes.Should().Contain(allChangeType);
```

Comment:
Can we use AutoFaker to create instances of the change types instead, so we don't have to remember to add each new one?

Reply:
yeah probably, not sure why I didn't think of that